### PR TITLE
 Copy button functionality

### PR DESCRIPTION
**Pull request by 0tuedon**
This PR Closes https://github.com/bitcoinsearch/chat-btc/issues/83

### CODE REVIEW

File: src/components/message/message.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import styles from "./message.module.css";
 import { LinkShareIcon } from "@/chakra/custom-chakra-icons";
 import { USER_REFERENCE_NAME } from "@/config/ui-config";
 import { separateLinksFromApiMessage } from "@/utils/links";
-import MarkdownWrapper from "./markdownWrapper/markdownWrapper";
+import MarkdownWrapper, { CopyButton, CopyResponseButton } from "./markdownWrapper/markdownWrapper";
```
Comment (Emmanuel-Develops) [inline src/components/message/message.tsx:8]: `CopyButton` is not being used here

File: src/components/message/message.tsx [hunk 2]
```diff
@@ -170,12 +170,19 @@ const MessageContent = ({
     separateLinksFromApiMessage(message);
 
   if (!messageBody?.trim()) return null;
-
+  console.log(type, messageBody)
```
Comment (Emmanuel-Develops) [inline src/components/message/message.tsx:173]: remove console

**Review by Emmanuel-Develops — CHANGES_REQUESTED** [review]

**Comment by Emmanuel-Develops** [comment]
The position of the copy button seems off.
Screenshot 2024-07-10 at 1 29 47 PM

Using other chat interfaces as guidelines, it is usually at the very bottom 

chatgpt:
Screenshot 2024-07-10 at 1 32 19 PM
claude:
Screenshot 2024-07-10 at 1 33 20 PM

try playing around with the placement

**Comment by 0tuedon** [comment]
> try playing around with the placement

I am trying to see because both examples don't have a source or follow-up questions.

doesn't displaying the copy button below make them think it's a copy-all situation?

File: src/components/message/markdownWrapper/markdownWrapper.tsx [hunk 3]
```diff
@@ -6,8 +6,12 @@ import Image from "next/image";
 import CopyIcon from "@/assets/CopyIcon";
 import { useState } from "react";
 import rehypeInlineCodeProperty from "../../../utils/rehypeInlineCodeProperty";
+import { useCopyToClipboard } from "usehooks-ts";
+import CopiedIcon from "@/assets/CopiedIcon";
+import { Box } from "@chakra-ui/react";
+import TickIcon from "@/assets/TickIcon";
 
-const CopyButton = ({ code }: { code: string }) => {
+export const CopyButton = ({ code }: { code: string }) => {
```
[the remark this answers is not in the document]
Comment (0tuedon) [reply src/components/message/markdownWrapper/markdownWrapper.tsx:14]: used in the `message.tsx`

**Comment by 0tuedon** [comment]
@Emmanuel-Develops 
Screenshot 2024-07-17 at 21 26 40
Screenshot 2024-07-17 at 21 23 42
at the very bottom

**Comment by Extheoisah** [comment]
> @Emmanuel-Develops Screenshot 2024-07-17 at 21 26 40 Screenshot 2024-07-17 at 21 23 42 at the very bottom

How about we just make the text "Copy" instead of "Copy Response"? Its shorter and it follows what others use

File: src/components/message/message.tsx [hunk 4]
```diff
@@ -217,6 +223,15 @@ const MessageContent = ({
           )}
         </Box>
       )}
+
+      {showCopyIcon && (
+        <Flex paddingY={"16px"} gap={2} alignItems={"center"} maxH={"max-content"} marginTop={"16px"}>
+             <Text fontSize="14px"  fontWeight={500}>
```
Comment (Extheoisah) [inline src/components/message/message.tsx:229]: can we make the text clickable too? so now when you click "copy", nothing happens. We should make the both the icon and text a clickable button

**Review by Extheoisah — APPROVED** [review]

**Pull request merged**